File: 01_counting_nucleotides/R/01_counting_nucleotides.R
#!/usr/bin/env Rscript

input <- readLines('stdin', n=1, warn=FALSE)

countBases <- function(base, seq) {
  n <- length(attr(gregexpr(pattern=base, text=seq)[[1]], "match.length"))
  return(n)
}

output <- list()

output[['A']] <- countBases("A", input)
output[['C']] <- countBases("C", input)
output[['G']] <- countBases("G", input)
output[['T']] <- countBases("T", input)

cat(c(output$A, output$C, output$G, output$T, '\n'))

File: 01_DNA_counting_nucleotides/python/01_counting_nucleotides.py
#!/usr/bin/env python

import sys


def main():
    seq = sys.stdin.readline().strip()
    nucleotide_counts = dict((base, seq.count(base)) for base in seq)

    print(" ".join(str(nucleotide_counts[x]) for x in ["A", "C", "G", "T"]))

if __name__ == '__main__':
    main()
